### PR TITLE
[BRE-101] Update CODEOWNERS for CI/CD Partnership Phase 2

### PR DESCRIPTION
**Pull request by michalchecinski**
Prepare CODEOWNERS for CI/CD partnership Phase 2:
- Docker files have joint ownership
- BRE owns `.github/workflows/publish.yml`
- Other workflows have joint ownership between bre and @bitwarden/team-auth-dev:
  - `.github/workflows/build.yml`
  - `.github/workflows/release.yml`
  - `.github/workflows/version-bump.yml`

https://bitwarden.atlassian.net/browse/BRE-101

### CODE REVIEW

**Comment by michalchecinski** [comment]
Wait for the https://bitwarden.atlassian.net/servicedesk/customer/portal/5/TSD-648 to be done.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
 # Multiple owners
 src/KeyConnector/KeyConnector.csproj
 .github/renovate.json
 Directory.Build.props
+
+## Docker files have shared ownership ##
+**/Dockerfile
+**/*.Dockerfile
+**/.dockerignore
+**/entrypoint.sh
```
Comment (vgrassia) [inline .github/CODEOWNERS:18]:
```suggestion
## Docker files have joint ownership ##
**/Dockerfile @bitwarden/dept-bre @bitwarden/team-auth-dev
**/*.Dockerfile @bitwarden/dept-bre @bitwarden/team-auth-dev
**/.dockerignore @bitwarden/dept-bre @bitwarden/team-auth-dev
**/entrypoint.sh @bitwarden/dept-bre @bitwarden/team-auth-dev
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -6,17 +6,18 @@
...
+
+## These are shared workflows ##
+.github/workflows/version-bump.yml
+.github/workflows/cleanup-rc-branch.yml
+.github/workflows/release.yml
```
Comment (vgrassia) [inline .github/CODEOWNERS:23]:
```suggestion
## BRE team owns these workflows ##
.github/workflows/publish.yml @bitwarden/dept-bre

## These workflows have joint ownership ##
.github/workflows/build.yml @bitwarden/dept-bre @bitwarden/team-auth-dev
.github/workflows/release.yml @bitwarden/dept-bre @bitwarden/team-auth-dev
.github/workflows/version-bump.yml @bitwarden/dept-bre @bitwarden/team-auth-dev
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Review by vgrassia — APPROVED** [review]

**Review by vgrassia — APPROVED** [review]

**Review by trmartin4 — APPROVED** [review]
As we've discussed, the Auth team has agreed to transition ownership of Key Connector to the KM team.

**Comment by quexten** [comment]
Seems this didn't require KM review to be moved into KM ownership. Still, looks good to me. Thank you for assigning ownership!

**Pull request merged**